test_vae_training/2_check_trained_model.py

```diff
@@ -0,0 +1,214 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:common_microbe] *
+#     language: python
+#     name: conda-env-common_microbe-py
+# ---
+
+# # Check training model
+#
+# This notebook will examine the trained model to see if it learned patterns of the input data. Specifically this notebook will look at:
+#
+# 1. Is there any structure in the data or just noise?
+# 2. Does our latent space capture the clusters in our input data? Are there samples that we know should cluster together? Do we find those in the input and encoded data?
+# 3. Do simulated samples, looks similar to the input data?
+
+# +
+# %load_ext autoreload
+# %autoreload 2
+# %matplotlib inline
+import os
+import glob
+
+# import umap
+import pandas as pd
+from keras.models import load_model
+from sklearn.decomposition import PCA
+from plotnine import (
+    ggplot,
+    labs,
+    geom_line,
+    geom_point,
+    geom_errorbar,
+    aes,
+    ggsave,
+    theme_bw,
+    theme,
+    xlim,
+    ylim,
+    facet_wrap,
+    scale_color_manual,
+    guides,
+    guide_legend,
+    element_blank,
+    element_text,
+    element_rect,
+    element_line,
+    coords,
+)
+from cm_modules import paths
+from ponyo import utils, simulate_expression_data
+
+random_state = 123
+
+# +
+base_dir = os.path.abspath(os.path.join(os.getcwd(), "../"))
+
+# Read in config variables
+config_filename = os.path.abspath(
+    os.path.join(base_dir, "test_vae_training", "config_current_vae.tsv")
+)
+
+params = utils.read_config(config_filename)
+
+dataset_name = params["dataset_name"]
+NN_architecture = params["NN_architecture"]
+normalized_compendium_filename = params["normalized_compendium_filename"]
+
+NN_dir = os.path.join(base_dir, dataset_name, "models", NN_architecture)
+# -
+
+# Import normalized data
+normalized_compendium = pd.read_csv(
+    normalized_compendium_filename, sep="\t", index_col=0, header=0
+)
+
+normalized_compendium.head()
+
+# Drop outliers found manually
+samples_to_drop = [
+    "PRJEB34610_ERR3561806",
+    "PRJEB34610_ERR3561830",
+    "PRJNA297268_SRR2568180",
+]
+normalized_compendium = normalized_compendium.drop(samples_to_drop)
+
+# ### Plot umap of normalized data
```

Comment:
Should be "Plot PCs of normalized data"

What happens when you run UMAP on microbiome data? If you didn't run the analysis you don't need to, I'm just curious what UMAP behaves like in zero-heavy data

Reply:
Great questions. I was actually trying to plot the umap of this too but I ran into some env issues with umap and decided to move forward with the count loss since its likely that that update will make the most difference. Though I'll try to return to this later.